packages/select/src/components/query-list/queryList.tsx

```diff
@@ -275,9 +291,11 @@ export class QueryList<T> extends React.Component<IQueryListProps<T>, IQueryList
         return getFirstEnabledItem(this.state.filteredItems, this.props.itemDisabled, direction, startIndex);
     }
 
-    private setActiveItem(item: T | undefined) {
-        // TODO: this will be more complex with un/controlled support (next PR)
-        Utils.safeInvoke(this.props.onActiveItemChange, item);
+    private setActiveItem(activeItem: T | undefined) {
+        if (this.props.activeItem == null) {
```

Comment:
only set state if it's null?

Reply:
if props is null then we're uncontrolled, so update state. otherwise inform the user and let them update props.